### ISSUE TITLE
Incremental compilation "sometimes" does not delete previous outputs

### ISSUE THREAD

**Issue by yigit** [google/ksp#332]
I'm trying to add incremental compilation to room and one of our existing tests is failing.
The test asserts that deleting a source element will delete the file that was generated for it. For some reason, Outputs to remove is empty in the `kspDirrtySetByOutputs` directory. Unfortunately, I've not been able to reproduce this outside AndroidX.

I'm *suspecting* that this might be some path issues. If you check the `kspDirtySetByOutputs` log, it has some of the files as absolute paths while others as local paths.

```
// from kspDirtySetByDeps.log
Removed
  src/main/java/room/testapp/Dao1.java:/tmp/junit2413537717527726914/src/main/java/room/testapp/Database1.java:/tmp/junit2413537717527726914/src/main/java/room/testapp/Entity1.java
Dirty
```
The compilation fails with:

```
> Task :compileDebugJavaWithJavac FAILED
/tmp/junit2413537717527726914/build/generated/ksp/debug/java/room/testapp/Database1_Impl.java:28: error: cannot find symbol
public final class Database1_Impl extends Database1 {
                                          ^
```
Database1 is deleted in the test but if you check `kspSourceToOutputs` below, we've set it as a dependency on `Database1_Impl`.

Here are the contents of the 4 incremental logs:

**kspDirtySet.log**
```
=== Build 1614283533349 ===
All Files
  src/main/java/placeholer.kt
  src/main/java/room/testapp/Entity1.java
  src/main/java/room/testapp/Database1.java
  src/main/java/room/testapp/Dao1.java
  src/main/java/room/testapp/Entity2.java
  src/main/java/room/testapp/Dao2.java
  src/main/java/room/testapp/Database2.java
Dirty:
  src/main/java/placeholer.kt
  src/main/java/room/testapp/Entity1.java
  src/main/java/room/testapp/Database1.java
  src/main/java/room/testapp/Dao1.java
  src/main/java/room/testapp/Entity2.java
  src/main/java/room/testapp/Dao2.java
  src/main/java/room/testapp/Database2.java

Dirty / All: 100.00%

=== Build 1614283538201 ===
All Files
  src/main/java/placeholer.kt
  src/main/java/room/testapp/Entity2.java
  src/main/java/room/testapp/Dao2.java
  src/main/java/room/testapp/Database2.java
Dirty:

Dirty / All: 0.00%
```

**kspDirtySetByDeps**
```
=== Build 1614283538201 ===
Modified
Removed
  src/main/java/room/testapp/Dao1.java:/tmp/junit2413537717527726914/src/main/java/room/testapp/Database1.java:/tmp/junit2413537717527726914/src/main/java/room/testapp/Entity1.java
Dirty
```

**kspDirtySetByOutputs**
```
=== Build 1614283538201 ===
Dirty sources
  src/main/java/room/testapp/Dao1.java:/tmp/junit2413537717527726914/src/main/java/room/testapp/Database1.java:/tmp/junit2413537717527726914/src/main/java/room/testapp/Entity1.java
Outputs to remove

```

**kspSourceToOutputs**
```
=== Build 1614283533349 ===
Accumulated source to outputs map
  src/main/java/room/testapp/Database1.java:
    build/generated/ksp/debug/java/room/testapp/Database1_Impl.java
    build/generated/ksp/debug/java/room/testapp/Dao1_Impl.java
  src/main/java/room/testapp/Database2.java:
    build/generated/ksp/debug/java/room/testapp/Dao2_Impl.java
    build/generated/ksp/debug/java/room/testapp/Database2_Impl.java

Reprocessed sources and their outputs
  src/main/java/room/testapp/Database1.java:
    build/generated/ksp/debug/java/room/testapp/Dao1_Impl.java
    build/generated/ksp/debug/java/room/testapp/Database1_Impl.java
  src/main/java/room/testapp/Database2.java:
    build/generated/ksp/debug/java/room/testapp/Dao2_Impl.java
    build/generated/ksp/debug/java/room/testapp/Database2_Impl.java

All reprocessed outputs
  build/generated/ksp/debug/java/room/testapp/Dao1_Impl.java
  build/generated/ksp/debug/java/room/testapp/Dao2_Impl.java
  build/generated/ksp/debug/java/room/testapp/Database1_Impl.java
  build/generated/ksp/debug/java/room/testapp/Database2_Impl.java

=== Build 1614283538201 ===
Accumulated source to outputs map
  src/main/java/room/testapp/Database1.java:
    build/generated/ksp/debug/java/room/testapp/Database1_Impl.java
    build/generated/ksp/debug/java/room/testapp/Dao1_Impl.java
  src/main/java/room/testapp/Database2.java:
    build/generated/ksp/debug/java/room/testapp/Dao2_Impl.java
    build/generated/ksp/debug/java/room/testapp/Database2_Impl.java

Reprocessed sources and their outputs

All reprocessed outputs


```

**Comment by yigit** [comment]
Can you try with this app?
I wrote a simple shell script to repro it so you should be able to reproduce it by going into the project and then running `./repro.sh`
[ksp-incremental-bug-repro.tar.gz](https://github.com/google/ksp/files/6045621/ksp-incremental-bug-repro.tar.gz)

**Comment by ting-yuan** [comment]
Fixed in #333.
